### PR TITLE
Add masterless state.highstate integration test

### PR DESCRIPTION
**Pull request by Ch3LL**
### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/43539

Might need to be rebased if https://github.com/saltstack/salt/pull/44613 is merged first.

### CODE REVIEW

**Comment by rallytime** [comment]
@Ch3LL There is a related test failure on this. Can you swing by and fix it up?

File: tests/support/case.py [hunk 1]
```diff
@@ -210,8 +210,10 @@ def run_cp(self, arg_str, with_retcode=False, catch_stderr=False):
         arg_str = '--config-dir {0} {1}'.format(self.get_config_dir(), arg_str)
         return self.run_script('salt-cp', arg_str, with_retcode=with_retcode, catch_stderr=catch_stderr)
 
-    def run_call(self, arg_str, with_retcode=False, catch_stderr=False):
-        arg_str = '--config-dir {0} {1}'.format(self.get_config_dir(), arg_str)
+    def run_call(self, arg_str, local=False, with_retcode=False, catch_stderr=False):
```
Comment (rallytime) [inline tests/support/case.py]: The `local` kwarg addition should go at the end of the list so we don't cause an API error. :)

File: tests/support/case.py [hunk 2]
```diff
@@ -549,11 +551,12 @@ def run_cp(self, arg_str, with_retcode=False, catch_stderr=False):
                                catch_stderr=catch_stderr,
                                timeout=60)
 
-    def run_call(self, arg_str, with_retcode=False, catch_stderr=False):
+    def run_call(self, arg_str, local=False, with_retcode=False, catch_stderr=False):
```
Comment (rallytime) [inline tests/support/case.py]: Same comment here.

**Review by rallytime — CHANGES_REQUESTED** [review]

**Comment by Ch3LL** [comment]
I rebased this branch with 2017.7 which had the correct changes for moving `local` to the end of the kwarg list on the run_call methods. So the files is now removed and shouldn't be an issue. Here's to the hope that the tests pass :)

**Review by rallytime — APPROVED** [review]

**Pull request merged**